importlib_resources/abc.py

```diff
@@ -1,9 +1,18 @@
 import abc
-from typing import BinaryIO, Iterable, Text
+from typing import Any, BinaryIO, Iterable, Iterator, NoReturn, Text
+from io import BufferedReader
+from typing import Union, Optional
+from os import PathLike
 
 from ._compat import runtime_checkable, Protocol
 
 
+__all__ = ["ResourceReader", "Traversable", "TraversableResources"]
+
+
+StrPath = Union[str, PathLike[str]]
```

Comment:
The `os.PathLike` subscripting was only added in 3.8 IIRC, so you need to change this definition based on the version.

Reply:
This should fix it, I think:

```suggestion
# PathLike is only subscriptable at runtime in 3.9+
StrPath = Union[str, "PathLike[str]"]
```

Reply:
I didn't understand this comment until I read the diff. I have an idea about how I'd prefer this Python-dependent variance to be handled.